addon/modifiers/did-resize.js

```diff
@@ -38,19 +56,30 @@ export default class DidResizeModifier extends Modifier {
     this.handler = handler;
     this.options = options || this.options;
 
-    this.observe();
+    // Only observe if the modifier is installed
+    if (this._didInstall) {
+      this.observe();
+    }
   }
 
   didInstall() {
     if (!('ResizeObserver' in window)) {
       return;
     }
 
-    this.observer = new ResizeObserver((entries, observer) => {
-      this.handler(entries[0], observer);
-    });
+    if (!DidResizeModifier.observer) {
+      DidResizeModifier.observer = new ResizeObserver((entries, observer) => {
+        for (let entry of entries) {
```

Comment:
This seems like a pretty expensive O(n^2) operation, having to iterate through all the entries and again through all of the handlers. Can this be simplified by using `this.element`?

```
for (let entry of entries) {
  if (this.element === entry.target) {
    this.handler(entry, observer);
  }
}
```

If not, maybe the proposed switch to WeakMap will simplify this logic somehow?